src/wrappers/themis/rust/src/secure_cell.rs

```diff
@@ -359,15 +683,79 @@ fn decrypt_token_protect(
     Ok(decrypted_message)
 }
 
-/// Secure Cell in a _context imprint_ operation mode.
+/// Secure Cell in _context imprint_ operation mode.
+///
+/// In this mode the input data is mixed with the provided context and encrypted, but there is no
+/// authentication token. Use this mode when you have no additional storage available for the
+/// authentication data and you absolutely need the output data to have the same length as the
+/// original input:
+///
+/// ```
+/// # fn main() -> Result<(), themis::Error> {
+/// use themis::secure_cell::SecureCell;
+///
+/// let cell = SecureCell::with_key_and_context(b"password", b"context").context_imprint();
+///
+/// let input = b"test input";
+/// let output = cell.encrypt(input)?;
+///
+/// assert!(output.len() == input.len());
+/// # Ok(())
+/// # }
+/// ```
+///
+/// Note that in context imprint mode you *must* provide non-empty context. Also keep in mind that
+/// Secure Cell cannot verify integrity and correctness of the decrypted data so you have to have
+/// some other means in place to validate the output.
 pub struct SecureCellContextImprint(SecureCell);
 
+// TODO: maybe panic if a SecureCell with an empty context is switched into context imprint mode
```

Comment:
must have

Reply:
Will do together with other API changes to Secure Cell. And we'll probably need some unit-tests and documentation updates ("Panics" sections) to communicate clearly that passing an empty context or a key is big no-no and you should never ever do this (or your application will crash).

We could leave the behavior as it is now (`encrypt`/`decrypt` returns an error when called for a Secure Cell with an empty key or context), but I think this warrants an early check and a loud failure. This is a programmer error and a data structure contract breach, like using an invalid index when addressing an array (which _does_ panic in Rust).

Though I'll update the documentation for context imprint mode in this PR.

Reply:
it totally makes sense to add panic at the early stage